.github/workflows/build-and-test.yaml

```diff
@@ -50,14 +50,23 @@ jobs:
         shell: bash
         run: |
           sudo apt-get update
-          sudo apt-get install -y cmake build-essential libfmt-dev
-          sudo apt-get install -y gperf build-essential bison flex libreadline-dev gawk tcl-dev libffi-dev git graphviz xdot pkg-config python3 libboost-system-dev libboost-python-dev libboost-filesystem-dev zlib1g-dev
+          sudo apt-get install -y git build-essential pkg-config cmake ccache gperf bison flex libreadline-dev gawk tcl-dev libffi-dev graphviz xdot python3 libboost-system-dev libboost-python-dev libboost-filesystem-dev zlib1g-dev
       - uses: actions/checkout@v4
         with:
           fetch-depth: 0
-      - name: Initialize submodules
+          submodules: recursive
+      - name: Set up caching
+        uses: actions/cache@v4
+        with:
+          path: ~/.cache/ccache
+          key: slang-${{ hashFiles('third_party') }}-${{ hashFiles('src') }}
+          restore-keys: |
```

Comment:
So we try to find as close match as we can find, but don't need an exact match, cool

Reply:
Also good call on replacing the previous ad hoc setup with ccache

Reply:
I think the previous ad hoc setup would have just not worked with cmake, so I had to. It is also a lot cleaner.